BridgeAppSDK/SBASurveyItem+Dictionary.swift

```diff
@@ -33,15 +33,36 @@
 
 import ResearchKit
 
+
+extension NSDictionary: SBAStepTransformer {
+    
+    // Because an NSDictionary could be used to create both an SBASurveyItem *and* an SBAActiveTask
+    // need to look to see which is the more likely form to result in a valid result.
+    public func transformToStep(factory: SBASurveyFactory, isLastStep: Bool) -> ORKStep {
+        if (self.surveyItemType.isNilType()) {
+            let taskOptions: ORKPredefinedTaskOption = isLastStep ? .None : .ExcludeConclusion
+            guard let subtask = factory.createTaskWithActiveTask(self, taskOptions:taskOptions) else {
+                return ORKStep(identifier: self.schemaIdentifier)
+            }
+            return SBASubtaskStep(subtask: subtask)
+        }
+        else {
+            return factory.createSurveyStep(self, isSubtaskStep: nil, isLastStep: isLastStep)
+        }
+    }
+}
+
 extension NSDictionary: SBASurveyItem {
     
     public var identifier: String! {
-        return (self["identifier"] as? String) ?? "\(self.hash)"
+        return (self["identifier"] as? String) ?? (self["schemaIdentifier"] as? String) ?? NSUUID().UUIDString
```

Comment:
different value every time you access this instance's identifier if no identifier or schemaIdentifier keys in the underlying dictionary